lib/src/Base/Func/FieldToFieldConnection.cxx

```diff
@@ -0,0 +1,205 @@
+}
+
+/* Parameters constructor */
+FieldToFieldConnection::FieldToFieldConnection(const PointToFieldFunction & pointToFieldFunction,
+                                       const FieldToPointFunction & fieldToPointFunction)
+  : FieldFunctionImplementation(fieldToPointFunction.getSpatialDimension(), fieldToPointFunction.getInputDimension(), fieldToPointFunction.getOutputDimension())
+  , isFieldFunctionComposition_(false)
+  , leftFieldFunction_()
+  , rightFieldFunction_()
+  , fieldToPointFunction_(fieldToPointFunction)
+  , pointToFieldFunction_(pointToFieldFunction)
+{
+  // Check if the dimensions of the point to field and field to point functions are compatible
+  if (fieldToPointFunction_.getInputDimension() != pointToFieldFunction_.getOutputDimension()) throw InvalidArgumentException(HERE) << "The input dimension=" << fieldToPointFunction_.getInputDimension() << " of the field to point function must be equal to the output dimension=" << pointToFieldFunction_.getOutputDimension() << " of the point to field function to compose them";
+  setInputDescription(fieldToPointFunction_.getInputDescription());
+  setOutputDescription(pointToFieldFunction_.getOutputDescription());
+}
+
+/* Virtual constructor */
+FieldToFieldConnection * FieldToFieldConnection::clone() const
+{
+  return new FieldToFieldConnection(*this);
+}
+
+/* Comparison operator */
+Bool FieldToFieldConnection::operator ==(const FieldToFieldConnection & other) const
+{
+  if (this == &other) return true;
+  return pointToFieldFunction_ == other.pointToFieldFunction_ && fieldToPointFunction_ == other.fieldToPointFunction_ && leftFieldFunction_ == other.leftFieldFunction_ && rightFieldFunction_ == other.rightFieldFunction_;
+}
+
+/* Get the i-th marginal function */
+FieldToFieldConnection::Implementation FieldToFieldConnection::getMarginal(const UnsignedInteger i) const
+{
+  if (i >= getOutputDimension()) throw InvalidArgumentException(HERE) << "Error: the index of a marginal function must be in the range [0, outputDimension-1]";
+  return getMarginal(Indices(1, i));
+}
+
+/* Get the function corresponding to indices components */
+FieldToFieldConnection::Implementation FieldToFieldConnection::getMarginal(const Indices & indices) const
+{
+  if (!indices.check(getOutputDimension())) throw InvalidArgumentException(HERE) << "The indices of a marginal function must be in the range [0, dim-1] and must be different";
+  if (isFieldFunctionComposition_) return new FieldToFieldConnection(leftFieldFunction_.getMarginal(indices), rightFieldFunction_);
+  return new FieldToFieldConnection(pointToFieldFunction_.getMarginal(indices), fieldToPointFunction_);
+}
+
+
+/* String converter */
+String FieldToFieldConnection::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << FieldToFieldConnection::GetClassName()
+      << " name=" << getName()
+      << " isFieldFunctionComposition=" << std::boolalpha << isFieldFunctionComposition_
```

Comment:
boolalpha is useless with our oss overload, see:
https://github.com/openturns/openturns/commit/c78a6228dcd9e262caeef024702971fdac5e4a4c